manifests/init.pp

```diff
@@ -313,7 +313,18 @@
   Optional[String] $agent_version = $datadog_agent::params::agent_version,
 ) inherits datadog_agent::params {
 
-  if $agent_major_version != 5 and $agent_major_version != 6 and $agent_major_version != 7 {
+  if $agent_version != 'latest' and $agent_version =~ /([0-9]+:)?([0-9]+)\.([0-9]+)\.([0-9]+)([^-\s]+)?(?:-([0-9]+))?/ {
+    $_agent_major_version = 0 + $2 # Cast to integer
+    if $agent_major_version != undef and $agent_major_version != $_agent_major_version {
+      fail('Provided and deduced agent_major_version don\'t match')
+    }
+  } elsif $agent_major_version != undef {
+    $_agent_major_version = $agent_major_version
+  } else {
+    $_agent_major_version = $datadog_agent::params::default_agent_major_version
```

Comment:
With this `else` statement, if `$agent_major_version` is `undef` and `$agent_version` is set to `6.15.0`, then `$_agent_major_version` will be equal to `7`, which is not expected. To get the correct behavior, I think the `else` statement should be removed. The `elsif` statement can then also be removed.

Reply:
Oh true, I misread the condition because the `elsif` was aligned with the inner `if`